### ISSUE TITLE
Stochastic error in plot.EventLogger

### ISSUE THREAD

**Issue by ArtPoon** [PoonLab/twt#123]
```R
> require(twt)
> path <- system.file('extdata', 'structSI.yaml', package='twt')
> mod <- Model$new(yaml.load_file(path))
> set.seed(42)
> run <- sim.outer.tree(mod)
> phy <- sim.inner.tree(run)
> plot(phy)
Error in `$<-.data.frame`(`*tmp*`, "lineage1", value = "Node49transmission1") : 
  replacement has 1 row, data has 0
```

**Comment by ArtPoon** [comment]
Running into a different problem on the Mac:
```R
> path <- system.file('extdata', 'structSI.yaml', package='twt')
> mod <- Model$new(yaml.load_file(path))
> set.seed(42)
> run <- sim.outer.tree(mod)
> plot(run)
> phy <- sim.inner.tree(run)
Error in private$sampling.times[names(which(private$locations == location))] <=  : 
  comparison (4) is possible only for atomic and list types
```

**Comment by ArtPoon** [comment]
Fixed in commit 689396e

**Comment by ArtPoon** [comment]
The original error here:
https://github.com/PoonLab/twt/blob/689396eb0e8b51d153e6cea953b5b877e30ee5db/R/EventLogger.R#L355-L382

Stepping through manually:
```R
> node <- unique(tm.events$lineage1)[16]
> node
[1] "Node49"
> # executing the above code
Error in `$<-.data.frame`(`*tmp*`, lineage1, value = "Node49transmission1") : 
  replacement has 1 row, data has 0
> new.node
[1] "Node49transmission1"
> e
[1] event.type   time         lineage1     lineage2     compartment1
[6] compartment2 type1        type2       
<0 rows> (or 0-length row.names)
```

`Node49` is the root of the inner tree.  There are no core events (coalescent or bottleneck) in which the root is the child node (`lineage1`). 

In this particular simulation, there is a transmission of the root lineage between unsampled hosts:
```R
> tail(tm.events)
     event.type     time lineage1 lineage2 compartment1 compartment2 type1 type2
47    migration 6.251344   Node39     <NA>  US_host2_40  US_host1_44 host2 host1
48 transmission 6.615025   Node37     <NA>  US_host2_39  US_host1_45 host2 host1
49 transmission 6.618134   Node42     <NA>  US_host2_43  US_host1_44 host2 host1
50 transmission 6.935030   Node45     <NA>  US_host2_40  US_host1_44 host2 host1
51 transmission 7.034434   Node46     <NA>  US_host1_45  US_host1_44 host1 host1
52 transmission 9.378565   Node49     <NA>  US_host1_44  US_host1_27 host1 host1
```
